### ISSUE TITLE
Refactoring

### ISSUE THREAD

**Issue by korvinos** [nansencenter/django-geo-spaas-sar-doppler#5]
I suggest a refactoring of the code to achieve  #3. If we "kiss" the code it will be easier to write tests for small functions and even possibly move some part of the code to the higher level (to reuse in other apps).

- [x] improve documentation within the code

- [x] make get_or_create only add the gsar rvl datasets to the db

- [x] add a method (post-) process to managers.py

**Comment by mortenwh** [comment]
I agree - this should definitely be done. Let's try to have meeting with
jeong won on Wednesday. Could you set it up?

9. des. 2017 11:01 skrev "Artem Moiseev" <notifications@github.com>:

I suggest a refactoring of the code to achieve #3
<https://github.com/nansencenter/django-geo-spaas-sar-doppler/issues/3>. If
we "kiss" the code it will be easier to write tests for small functions and
even possibly move some part of the code to the higher level (to reuse in
other apps).

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/nansencenter/django-geo-spaas-sar-doppler/issues/5>,
or mute
the thread

**Comment by korvinos** [comment]
@mortenwh Yes, sure

**Comment by mortenwh** [comment]
I have done the above tasks in the master branch now. We need to consider how to handle the way of dealing with figures, e.g., as in the development branch. The `get_or_create` method needs to be called in `process` to ensure the gsar file is added as a dataset. Apart from that, `process` generates the geophysical Doppler shift and creates visualizations for the browser view. You can see a usage example in `svp_drifters_demo.py` in the sar-doppler repos.
